src/services/codefixes/helpers.ts

```diff
@@ -40,7 +40,7 @@ namespace ts.codefix {
         const scriptTarget = getEmitScriptTarget(context.program.getCompilerOptions());
         const declaration = declarations[0];
         const name = getSynthesizedDeepClone(getNameOfDeclaration(declaration), /*includeTrivia*/ false) as PropertyName;
-        const visibilityModifier = createVisibilityModifier(getModifierFlags(declaration));
+        const visibilityModifier = createVisibilityModifier(getSyntacticModifierFlags(declaration));

```

Comment:
this function only works for TS, right? Otherwise we'd need to accept declaration with JSDoc modifiers and produce a node with JSdoc annotations.

Reply:
Its possible this could be used when implementing an interface inferred from a JS class:
```ts
// foo.js
export class C { /** @readonly */ x = 1; }
// bar.ts
import { C } from "./foo";
class D implements C { // <- code fix to implement interface should add `readonly`
}
```

I'll switch this to use `getEffectiveModifierFlags`.